### ISSUE TITLE
Improve the way we handle dates and events for multi-day events

### ISSUE THREAD

**Issue by alexwlchan** [wellcomecollection/wellcomecollection.org#8830]
**tl;dr: We aren't handling the date ranges on Lights Up properly. In November it's just a bit wonky, but in December the event will completely vanish from the site unless we do something.**

---

The new exhibition has a series of [Lights Up sessions](https://wellcomecollection.org/events/YzGDAxEAAM-gfxhj), which occurs seven times between 3 Nov and 9 Feb. These are breaking the site in fun and new ways, because AFAIK this is the first time we've had an event that spans more than two months.

I've spotted at least three bugs with Lights Up:

* Lights Up doesn't appear anywhere on the [global events page](https://wellcomecollection.org/events), because it's neither fully present or fully past
* Lights Up also doesn't appear in the December tab of the What's on page, even though it occurs twice that month. I think it will disappear after December.
* It doesn't sort properly in the list of What's on. The next Lights Up event isn't until November 15th, but it's appearing before events on November 8th and November 10th. It will continue to appear in November even when the next event isn't until December.

And I think the card is a bit confusing:

<img width="639" alt="Screenshot 2022-11-04 at 14 04 56" src="https://user-images.githubusercontent.com/301220/199992589-beb32bc6-cd04-4c83-9964-922e7178a1d0.png">

It has a continuous range of dates, which looks like how we present exhibition cards – you can wander up any time during those dates and see it (no!). The next Lights Up event is actually on November 15th.

To my eye, the HIV and AIDS card is much clearer that it's an event which happens on discrete dates at multiple times.

# The Prismic data

I think the root of this issue is how we manage the data about event times, and what we have available where when we render the front-end.

This is how "Lights Up" is modelled in Prismic:

```mermaid
graph LR
    L["<b>Lights Up</b><br/>times: 3 Nov &ndash; 9 Feb"]
    L --> L1["<b>Lights Up session 3 Nov</b><br/>times: 3 Nov @ 4pm &ndash; 8pm<br/>delist = true"]
    L --> L2["<b>Lights Up session 15 Nov</b><br/>times: 15 Nov @ 10am &ndash; 2pm<br/>delist = true"]
    L --> L3["<b>Lights Up session 6 Jan</b><br/>times: 6 Jan @ 2pm &ndash; 6pm<br/>delist = true"]

    classDef eventNode text-align:left,fill:#fff9e6,stroke:#ffebad
    class L,L1,L2,L3 eventNode

    classDef mainEvent stroke-width:3,fill:#ffebad,stroke:#ffce3c
    class L mainEvent
```

There's one event which appears on the What's on page, which represents the group of events. There are then individual events for each of the sessions, which are linked via the Schedule tab in Prismic. These events are delisted, so they don't appear on the what's on page.

The Editorial team will put:

* the times for each event on the individual Event document in Prismic
* a range from the earliest start to the latest end on the summary Event document

When we render promo cards, we only have access to that summary data.

# Proposed changes

If we could get the individual event dates when we render promo cards, I think we could do a better job of handling this and similar events. We'd be able to:

* Highlight the next time the event occurs
* Make it clear this is a discrete series of events, not something continuously running
* Sort it in the right place on the "What's on" page

Here's a little mockup I did of a possible alternative design:

<img width="636" alt="Screenshot 2022-11-04 at 14 17 50" src="https://user-images.githubusercontent.com/301220/199997187-906f6aed-00f2-4245-9949-f428ed1bf838.png">

We might be able to get this data by changing the way we query Prismic, or we could change the way the data is stored in Prismic. (But I don't know what the implications of that are.)

**Comment by alexwlchan** [comment]
We've fixed this in the short-term, but there's a bigger question here about whether we're modelling these events in Prismic.

In particular, each sub-event is a full event, even though they can't be linked or viewed as standalone events.

I'm going to leave this ticket open to track this content modelling.
